docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,493 @@
+
+.. sourcecode:: json
+
+   $ cat src/main/resources/avro/io/confluent/examples/clients/basicavro/Payment.avsc
+   {"namespace": "io.confluent.examples.clients.basicavro",
+    "type": "record",
+    "name": "Payment",
+    "fields": [
+        {"name": "id", "type": "string"},
+        {"name": "amount", "type": "double"}
+    ]
+   }
+
+Let's break down what this schema defines
+
+* ``namespace``: a fully qualified name that avoids schema naming conflicts
+* ``type``: `Avro data type <https://avro.apache.org/docs/1.8.1/spec.html#schemas>`_, one of `record`, `enum, `union`, `array`, `map`, `fixed`
+* ``name``: unique schema name in this namespace
+* ``fields``: one or more simple or complex data types for a `record`. The first field in this record is called `id`, and it is of type `string`. The second field in this record is called `amount`, and it is of type `double`.
+
+
+Client Applications Writing Avro
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Apache Kafka applications using Avro data and |sr-long| need to specify at least two configuration parameters:
+
+* Avro serializer or deserializer
+* URL to the |sr-long|
+
+Java Producers
+^^^^^^^^^^^^^^
+
+Java applications that have Kafka producers using Avro require ``pom.xml`` files to include:
+
+* Avro dependencies to serialize data as Avro, including ``org.apache.avro.avro`` and ``io.confluent.kafka-avro-serializer``
+* Avro plugin ``avro-maven-plugin`` to generate Java class files from the source schema
+
+For a full pom.xml example, refer to this `pom.xml <https://github.com/confluentinc/examples/blob/5.0.0-post/clients/avro/pom.xml>`_.
+
+Within the application, Java producers that are serializing data as Avro set two main configurations parameters:
+
+* Avro serializer for the Kafka value (or Kafka key)
+* URL to the |sr-long|
+
+Then the producer can write records where the Kafka value is of `Payment` class.
+For example:
+
+.. sourcecode:: java
+
+   import io.confluent.kafka.serializers.KafkaAvroSerializer;
+   import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
+
+   ...
+   props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
+   props.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, schemaRegistryUrl);
+   ...
+
+   ...
+   KafkaProducer<String, Payment> producer = new KafkaProducer<String, Payment>(props));
+   final Payment payment = new Payment(orderId, 1000.00d);
+   final ProducerRecord<String, Payment> record = new ProducerRecord<String, Payment>(TOPIC, payment.getId().toString(), payment);
+   producer.send(record);
+   ...
+
+For a full Java producer example, refer to `the producer example <https://github.com/confluentinc/examples/blob/5.0.0-post/clients/avro/src/main/java/io/confluent/examples/clients/basicavro/ProducerExample.java>`_.
+
+
+Java Consumers
+^^^^^^^^^^^^^^
```

Comment:
Thanks! Sorry for all the duplicate comments! Github was having issues last night, and I wasn't seeing them being submitted.

Since this tutorial focuses on Specific, it might help to mention a bit more about the plugin. Specifically, `mvn compile` can be used to generate the classes, which will be written into the `target/generated-sources` folder. Otherwise, it doesn't seem immediately clear where the Kafka value object type comes from. 

Also, I'm a fan of AVDL over AVSC, but maybe that's duplicating too much information from the Avro docs, and that decision is up to the implementor.

Reply:
> Leaving as AVSC. The goal of this tutorial is to show one path that is easiest to understand

Understood. I'll PR into the examples repo at some time to iterate the alternatives w.r.t. the avro-plugin.

Reply:
Sounds great!